unixfs/mod/dagmodifier.go

```diff
@@ -460,7 +460,7 @@ func dagTruncate(nd *mdag.Node, size uint64, ds mdag.DAGService) (*mdag.Node, er
 	var modified *mdag.Node
 	ndata := new(ft.FSNode)
 	for i, lnk := range nd.Links {
-		ctx, cancel := context.WithTimeout(context.TODO(), time.Minute)
+		_ctx, cancel := context.WithTimeout(ctx, time.Minute)
```

Comment:
maybe this timeout one is one to remove? 

We'd still want a context derived from the one passed in, so things can be cancelled when this func returns